src/main/java/com/xeeva/catalog/ConfirmRequestPage.java

```diff
@@ -1,98 +1,286 @@
+		TestReporter.logStep("Grand Total Tax : "+ActualGrandTotalTax[0]);
+
+		return ActualGrandTotalTax[0];
+
+	}
+
+	/**
+	 * @summary Method to calculate Grand total 
+	 * @author  Lalitha Banda
+	 * @date    29/09/16
+	 */
+	public void verifyGrandTotal(){
+		String subTotal = verifySubTotal();
+		String grandTotalTax =  verifyGrandTotalTax();
+
+		/**Calculate Grand total by adding Sub total and Grand total tax */
+		double grandTotal = Double.parseDouble(subTotal)+Double.parseDouble(grandTotalTax);
+		String ExpectedGrandTotal = new BigDecimal(String.valueOf(grandTotal)).setScale(5, BigDecimal.ROUND_HALF_UP).toString();
+
+		// Reading Grand Total
+		String[] ActualGrandTotal = lstTotalValues.get(2).getText().split(" ");
+
+		// Verifying  Grand Total 
+		TestReporter.assertTrue(ExpectedGrandTotal.equals(ActualGrandTotal[0]), "Grand Total available as sum of sub total and grand total tax");
+		TestReporter.logStep("Grand Total : "+ExpectedGrandTotal);
+
+	}
+
+	/**
+	 * @Summary: Method to click on Back button.
+	 * @author: Praveen Namburi, @Version: Created 30-09-2016
+	 */
+	public void clickbtnBack_ConfirmRequestPage(){
+		pageLoaded();
+		pl.isDomComplete(driver);
+		btnBack.syncEnabled(30);
+		driver.executeJavaScript("arguments[0].click();", btnBack);
+		driver.manage().timeouts().implicitlyWait(Constants.PAGE_TIMEOUT, TimeUnit.SECONDS);
+	}
+	
+	/**
+	 * @summary Method to click on checkbox- either Budget/Non-Budget and confirm the process.
+	 * @author  Praveen Namburi, @Version: Created 03-10-2016
+	 */
+	public void clickbtnConfirm(){
+		pl.isDomComplete(driver);
+		if(chkBudgeted.isDisplayed()){
+			click_Budgeted();
+		}else if (chkNonBudgeted.isDisplayed()) {
+			click_NonBudgeted();
+		}else {
+			btnConfirm.syncVisible(5, false);
+			driver.executeJavaScript("arguments[0].click();", btnConfirm);
+			driver.manage().timeouts().implicitlyWait(Constants.PAGE_TIMEOUT, TimeUnit.SECONDS);
+		}
+		
+	}
+
+	/**
+	 * @summary Method to attach a file 
+	 * @author  Lalitha Banda
+	 * @date    03/10/16
+	 */
+	public boolean attachfile(WebElement Element,String filename){
+		boolean statusFlag = false;
+
+		pageLoaded();
+		driver.executeJavaScript("arguments[0].click();", lstAttachFile.get(0));
+		driver.executeJavaScript("arguments[0].click();", Element);
+		btnBrowse.click();
+		try {Thread.sleep(3000);}catch (InterruptedException e){ e.printStackTrace();}
+
+		String fileLocation = System.getProperty("user.dir") + "\\src\\test\\resources\\datasheets\\" + filename+".xlsx";
+		TestReporter.logStep(fileLocation);
+
+		StringSelection filepath = new StringSelection(fileLocation);
+		Toolkit.getDefaultToolkit().getSystemClipboard().setContents(filepath, null);
+
+		try {
+			Robot robot = new Robot();
+			robot.keyPress(KeyEvent.VK_CONTROL);
+			robot.keyPress(KeyEvent.VK_V);
+			robot.keyRelease(KeyEvent.VK_V);
+			robot.keyRelease(KeyEvent.VK_CONTROL);
+			robot.keyPress(KeyEvent.VK_ENTER);
+			robot.keyRelease(KeyEvent.VK_ENTER);
+
```

Comment:
We will need to have a conversation about using the Robot class and a better strategy for this.  While this works locally on your machine - will not work when we execute these as a suite of test on the Jenkins machine - as the key presses will be on the jenkins machine and not done on the remote machines.